docs/site.md

```diff
@@ -30,15 +34,109 @@ You will also need to configure your repository settings:
 2. Set the GitHub pages source to the `/` (root) directory on the `gh-pages` branch.
   `https://github.com/{user}/{repo}/settings/pages`
 
+
+## Configuration
+
+If you run the plugin with its defaults it will generate a site that will look like this documentation.
+
+Below we'll describe how the default settings differ from Laika standalone
+as well as a few pointers for the most relevant customization options.
+
+
+### Site Default Settings
+
+Whereas Laika standalone is a general purpose tool, this plugin's domain is project documentation
+which allows us to make a few additional assumptions and add useful defaults based on those.
+
+On top of the defaults of Laika standalone, sbt-typelevel adds:
+
+* GitHubFlavor for Markdown is enabled by default (e.g. for fenced code blocks).
+* Laika's builtin syntax highlighting is enabled by default (which does not require JavaScript highlighters).
+* Metadata is pre-populated based on sbt settings (e.g. title, author, version).
+* A link to the GitHub repository is inserted into the top navigation bar based on the output of `git ls-remote --get-url`.
+* If you define the `tlSiteApiUrl` setting, a link to the API documentation is inserted into the top navigation bar
+  and a redirect for `/api/` to that URL will be added to your site.
+
+
+#### Additional Defaults for Typelevel Projects
+
+If the generated documentation is for a Typelevel project, you can optionally enable a set of additional defaults
+on top of the generic defaults listed in the previous section:
+
+```scala
+tlSiteIsTypelevelProject := true
+```
+
+With the flag above (which defaults to `false`) these additional settings apply:
+
+* The home link in the top navigation bar carries the Typelevel logo and points to the Typelevel site.
+* Links to the Typelevel Discord and Typelevel Twitter are inserted into the top navigation bar.
+* The Typelevel favicons are used for the generared site.
+* A default footer for Typelevel projects is added to the bottom of each page.
+* Theme support for the browser's dark mode is disabled.
```

Comment:
IIRC there was no interest in dark mode when we discussed my original PR for the http4s site. So I had to disable it, because http4s has a custom colour scheme, and you need to explicitly define the dark mode colours in such a case (otherwise users with dark mode switched on would suddenly see the default dark mode colours).

I think we can re-enable this in the plugin and only disable it for http4s until you have dark mode colours defined for it.

Reply:
Ah, I think I disabled it because the Typelevel logo at the top didn't look so good with the default dark mode color scheme. Unless there's an easy fix we should just leave it I think.

No pressure ... but would you be interested in making a custom color scheme for Typelevel, maybe to match the [redesigned website](https://typelevel.org/)? You did a really beautiful job with http4s 😇 if you have time for this, it can be a follow-up PR.

Reply:
I can give this a try. It looks like the Typelevel site does not really have a secondary color, so I'd maybe try something like with http4s, white/greyish/black, but with red as link/headline colour instead of orange?

Reply:
Yup, that sounds great!